src/main/main.ts

```diff
@@ -26,122 +27,132 @@ let appIsQuitting = false;
+        }
+        mainWindow.show();
+        mainWindow.focus();
+    });
+
+    mainWindow.on('minimize', (event: any) => {
+        event.preventDefault();
+        mainWindow?.hide();
+    });
+
+    mainWindow.on('close', (event) => {
+        if (!appIsQuitting) {
+            event.preventDefault();
+            mainWindow?.hide();
+            return false;
+        }
+        mainWindow?.destroy();
 
-  const menuBuilder = new MenuBuilder(mainWindow);
-  menuBuilder.buildMenu();
+        return true;
+    });
+
+    const menuBuilder = new MenuBuilder(mainWindow);
+    menuBuilder.buildMenu();
 
-  // Open urls in the user's browser
-  mainWindow.webContents.setWindowOpenHandler((edata) => {
-    shell.openExternal(edata.url);
-    return { action: 'deny' };
-  });
+    // Open urls in the user's browser
+    mainWindow.webContents.setWindowOpenHandler((edata) => {
+        shell.openExternal(edata.url);
+        return { action: 'deny' };
+    });
 
-  // Remove this if your app does not use auto updates
-  // eslint-disable-next-line
-  new AppUpdater();
+    // Remove this if your app does not use auto updates
+    // eslint-disable-next-line
+    new AppUpdater();
 };
 
 /**
  * Add event listeners...
  */
 app.on('window-all-closed', () => {
-  // app.quit();
+    // app.quit();
 });
 
 app
-  .whenReady()
-  .then(() => {
-    createWindow();
-    const store = createStore();
-
-    // Register Shortcuts
-    if (!app.isPackaged) {
-      // Ctrl + Q to quit
-      globalShortcut.register('CommandOrControl+Q', () => {
-        appIsQuitting = true;
-        app.quit();
-      });
-    }
-
-    // Register Tray Icon
-    buildTray(mainWindow, RESOURCES_PATH, app.getVersion());
-
-    // TODO: Currently the API key is set manually by opening the config.json file
-    if (store.get('apiKey')) {
-      setupSignalR(store, createAlertsWindow);
-    }
-
-    app.on('activate', () => {
-      // On macOS it's common to re-create a window in the app when the
-      // dock icon is clicked and there are no other windows open.
-      if (mainWindow === null) createWindow();
-    });
-  })
-  .catch(console.log);
+    .whenReady()
+    .then(() => {
+        createWindow();
+        const store = createStore();
+
+        // Register Shortcuts
+        if (!app.isPackaged) {
+            // Ctrl + Q to quit
+            globalShortcut.register('CommandOrControl+Q', () => {
+                appIsQuitting = true;
+                app.quit();
+            });
+        }
+
+        // Register Tray Icon
+        buildTray(mainWindow, RESOURCES_PATH, app.getVersion());
+
+        // TODO: Currently the API key is set manually by opening the config.json file
+        if (store.get('apiKey')) {
+            setupSignalR(store, createAlertsWindow);
+        }
+
+        const autoav = new Worker(AUTOAV_BACKGROUND_THREAD_PATH);
+
+        autoav.on('message', (msg: string) => {
+            console.log('From AutoAV:', msg);
```

Comment:
Use logger?

Reply:
fixed, using an external logger as added in be5107a